src/series/bar.js

```diff
@@ -106,6 +106,17 @@ export default function() {
         });
     };
 
+    bar.width = function(data, orientation) {
```

Comment:
With this change it means that the bar exposes a `width` function that has different logic for calculating the widths than it uses itself. The two really should be unified.